Can't seem to `npm install`

I get the following error when attempting to `npm install`. Does anybody know why this is or what's going on? I don't have much experience with node.
 
    npm WARN package.json color-parser@0.1.0 No repository field.
     /
     > libxmljs@0.14.3 install /home/shaan/Desktop/Colorsublime-Themes/node_modules/libxmljs
     > node-gyp rebuild
    
    /bin/sh: 1: node: not found
    gyp: Call to 'node -e "require('nan')"' returned exit status 127. while trying to load binding.gyp
    gyp ERR! configure error 
    gyp ERR! stack Error: `gyp` failed with exit code: 1
    gyp ERR! stack     at ChildProcess.onCpExit (/usr/share/node-gyp/lib/configure.js:344:16)
    gyp ERR! stack     at ChildProcess.EventEmitter.emit (events.js:98:17)
    gyp ERR! stack     at Process.ChildProcess._handle.onexit (child_process.js:797:12)
    gyp ERR! System Linux 3.19.0-26-generic
    gyp ERR! command "nodejs" "/usr/bin/node-gyp" "rebuild"
    gyp ERR! cwd /home/shaan/Desktop/Colorsublime-Themes/node_modules/libxmljs
    gyp ERR! node -v v0.10.25
    gyp ERR! node-gyp -v v0.12.2
    gyp ERR! not ok 
    npm WARN This failure might be due to the use of legacy binary "node"
    npm WARN For further explanations, please read
    /usr/share/doc/nodejs/README.Debian
     
    npm ERR! libxmljs@0.14.3 install: `node-gyp rebuild`
    npm ERR! Exit status 1
    npm ERR! 
    npm ERR! Failed at the libxmljs@0.14.3 install script.
    npm ERR! This is most likely a problem with the libxmljs package,
    npm ERR! not with npm itself.
    npm ERR! Tell the author that this fails on your system:
    npm ERR!     node-gyp rebuild
    npm ERR! You can get their info via:
    npm ERR!     npm owner ls libxmljs
    npm ERR! There is likely additional logging output above.
    
    npm ERR! System Linux 3.19.0-26-generic
    npm ERR! command "/usr/bin/nodejs" "/usr/bin/npm" "install"
    npm ERR! cwd /home/shaan/Desktop/Colorsublime-Themes
    npm ERR! node -v v0.10.25
    npm ERR! npm -v 1.4.21
    npm ERR! code ELIFECYCLE
    npm ERR! 
    npm ERR! Additional logging details can be found in:
    npm ERR!     /home/shaan/Desktop/Colorsublime-Themes/npm-debug.log
    npm ERR! not ok code 0

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,6 +1,6 @@
 language: node_js
 node_js:
-  - 0.10.32
+  - "0.12"
 
 branches:
   only:

--- a/package.json
+++ b/package.json
@@ -22,9 +22,9 @@
   },
   "homepage": "http://colorsublime.com",
   "devDependencies": {
-    "chai": "^2.1.0",
+    "chai": "^3.5.0",
     "color-parser": "^0.1.0",
-    "libxmljs": "^0.14.1",
+    "libxmljs": "^0.18.0",
     "mocha": "^2.1.0",
     "plist": "^1.1.0",
     "walk": "^2.3.9"

